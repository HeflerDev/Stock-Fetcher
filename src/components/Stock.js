import React from 'react';
import PropTypes from 'prop-types';

const Stock = ({ stock }) => {
  const {
    symbol,
    name,
    currency,
    stockExchange,
    exchangeShortName,
  } = stock;

  return (
    <div className="stack">
      <ul className="board">
        <li className="col-12 big">
          <div>{ name }</div>
        </li>
        <li className="col-3">
          <div>{ symbol }</div>
        </li>
        <li className="col-3">
          <div>{ currency }</div>
        </li>
        <li className="col-3">
          { stockExchange }
        </li>
        <li className="col-3">
          { exchangeShortName }
        </li>
      </ul>
    </div>
  );
};

Stock.propTypes = {
  stock: PropTypes.shape({
    symbol: PropTypes.string.isRequired,
    name: PropTypes.string,
    currency: PropTypes.string.isRequired,
    stockExchange: PropTypes.string,
    exchangeShortName: PropTypes.string,
  }).isRequired,
};

Stock.defaultProps = {
  stock: {
    name: 'None',
  },
}

export default Stock;
