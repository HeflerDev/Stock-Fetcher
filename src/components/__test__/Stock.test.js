import renderer from 'react-test-renderer';
import Stock from '../Stock';

describe('when loading stock', () => {
  const mockProp = {
    symbol: 'symbol',
    name: 'name',
    currency: 'currency',
    exchangeShortName: 'ShortName',
  };

  test('it render correctly', () => {
    const tree = renderer
      .create(<Stock stock={mockProp} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
