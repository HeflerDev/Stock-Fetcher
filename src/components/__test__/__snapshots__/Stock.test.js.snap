// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading stock it render correctly 1`] = `
<div
  className="stack"
>
  <ul
    className="board"
  >
    <li
      className="col-12 big"
    >
      <div>
        name
      </div>
    </li>
    <li
      className="col-4"
    >
      <div>
        symbol
      </div>
    </li>
    <li
      className="col-4"
    >
      <div>
        currency
      </div>
    </li>
    <li
      className="col-4"
    >
      ShortName
    </li>
  </ul>
</div>
`;
